core/src/main/java/org/elasticsearch/index/shard/DocsStats.java

```diff
@@ -57,16 +66,26 @@ public long getDeleted() {
         return this.deleted;
     }
 
+    public long getAverageSizeInBytes() {
+        return averageSizeInBytes;
+    }
+
     @Override
     public void readFrom(StreamInput in) throws IOException {
         count = in.readVLong();
         deleted = in.readVLong();
+        if (in.getVersion().onOrAfter(Version.V_6_1_0)) {
```

Comment:
you might want to make it `Version.V_7_0_0` for now and change it back after this change is backported in order not to cause failures in the multi-version cluster qa tests

Reply:
Thanks for the hint. I temporarily made this for v7 only (https://github.com/elastic/elasticsearch/pull/27117/commits/f38e9578f7dadcf63cc0bc3097cd1807b8b7ba99)